pkg/kube/controllers/boshdeployment/bpm_reconciler.go

```diff
@@ -114,32 +117,43 @@ func (r *ReconcileBPM) Reconcile(request reconcile.Request) (reconcile.Result, e
 			log.WithEvent(bpmSecret, "LabelMissingError").Errorf(ctx, "Missing container label for bpm information bpmSecret '%s'", request.NamespacedName)
 	}
 
-	resources, err := r.applyBPMResources(bpmSecret, manifest)
-	if err != nil {
-		return reconcile.Result{},
-			log.WithEvent(bpmSecret, "BPMApplyingError").Errorf(ctx, "Failed to apply BPM information: %v", err)
-	}
-
-	if resources == nil {
-		log.WithEvent(bpmSecret, "SkipReconcile").Infof(ctx, "Skip reconcile: BPM resources not found")
-		return reconcile.Result{}, nil
-	}
-
 	// Start the instance groups referenced by this BPM secret
 	instanceName, ok := bpmSecret.Labels[bdv1.LabelDeploymentName]
 	if !ok {
 		return reconcile.Result{},
 			log.WithEvent(bpmSecret, "LabelMissingError").Errorf(ctx, "Missing deployment mame label for bpm information bpmSecret '%s'", request.NamespacedName)
 	}
 
-	// Deploy instance groups
 	instance := &bdv1.BOSHDeployment{}
 	err = r.client.Get(ctx, types.NamespacedName{Namespace: request.Namespace, Name: instanceName}, instance)
 	if err != nil {
 		return reconcile.Result{},
 			log.WithEvent(bpmSecret, "GetBOSHDeployment").Errorf(ctx, "Failed to get BoshDeployment instance '%s': %v", instanceName, err)
 	}
 
+	dns := manifest.DNS(request.Namespace)
```

Comment:
That is  a good point. We will refactor  the DNSPolicy  used, so that we should be able to use different DNS objects. We hopefully upload another version tomorrow.

Reply:
done: https://github.com/cloudfoundry-incubator/cf-operator/pull/598/commits/f7115851d84ea77f0bc2b226df7794d92265eb5e